test_submission_schema/v1.json

```diff
@@ -0,0 +1,167 @@
+                "type": "object",
+                "required": [
+                    "id",
+                    "test_description",
+                    "certification_status",
+                    "category_id",
+                    "outcome",
+                    "comments",
+                    "io_log",
+                    "duration",
+                    "type",
+                    "template_id"
+                ],
+                "properties": {
+                    "id": {
+                        "type": "string",
+                        "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[a-zA-Z0-9_]+/[a-zA-Z0-9_]+$",
+                        "description": "The unique identifier of the test case."
+                    },
+                    "test_description": {
+                        "type": "string",
+                        "description": "The description of the test case.",
+                        "maxLength": 80
+                    },
+                    "certification_status": {
+                        "type": "string",
+                        "description": "The certification status of the test case.",
+                        "enum": [
+                            "blocker",
+                            "non-blocker"
+                        ]
+                    },
+                    "category_id": {
+                        "type": "string",
+                        "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[A-Za-z0-9_\\-]+$",
+                        "description": "The unique identifier of the test case category."
+                    },
+                    "outcome": {
+                        "type": "string",
+                        "description": "The outcome of the test case.",
+                        "enum": [
+                            "passed",
+                            "failed",
+                            "skipped"
+                        ]
+                    },
+                    "comments": {
+                        "type": "string",
+                        "description": "Any additional comments or observations about the test case."
+                    },
+                    "io_log": {
+                        "type": "string",
+                        "description": "The stdout and stderr for the test case."
+                    },
+                    "duration": {
+                        "type": "number",
+                        "minimum": 0,
+                        "description": "The duration of the test case in milliseconds."
+                    },
+                    "type": {
+                        "type": "string",
+                        "description": "The type of test case.",
+                        "enum": []
+                    },
+                    "template_id": {
+                        "type": "string",
+                        "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[a-zA-Z0-9_]+/[a-zA-Z0-9_]+$",
```

Comment:
After implementing the converter in yarf, I feel like the `template_id` should not follow this pattern. For example, in robot framework, the templates are by keyword, not by test case. We can have a test case that has a number of keywords and all of them have their templates. I think in that case we should add them all here. Thoughts? @p-gentili @Hook25

Reply:
Like here: 
https://github.com/canonical/yarf/pull/105/files#diff-17f48944656db91e6df0b1b0ed4d3341568ec6b5aa75730fd0e344ffdb6ab3faR717-R759